Module1.js

```diff
@@ -2,213 +2,170 @@
+  
+	const registrySheet = SpreadsheetApp.openById(AMBASSADOR_REGISTRY_SPREADSHEET_ID).getSheetByName(REGISTRY_SHEET_NAME); // Open the "Registry" sheet
 	Logger.log('Opened "Registry" sheet from "Ambassador Registry" spreadsheet.');
-
-	const emails = registrySheet
-		.getRange(2, 1, registrySheet.getLastRow() - 1, 1)
-		.getValues()
-		.flat(); // Fetch all emails from the Registry sheet
+  
+	const emails = registrySheet.getRange(2, 1, registrySheet.getLastRow() - 1, 1).getValues().flat(); // Fetch all emails from the Registry sheet
 	Logger.log(`Fetched emails from "Registry" sheet: ${emails}`);
-
+  
 	const deliverableDate = getPreviousMonthDate(spreadsheetTimeZone); // Call from SharedUtilities.gs
 	Logger.log(`Deliverable date: ${deliverableDate}`);
-
-	const month = Utilities.formatDate(
-		deliverableDate,
-		spreadsheetTimeZone,
-		"MMMM"
-	); // Format the deliverable date to get the month name
-	const year = Utilities.formatDate(
-		deliverableDate,
-		spreadsheetTimeZone,
-		"yyyy"
-	); // Format the deliverable date to get the year
+  
+	const month = Utilities.formatDate(deliverableDate, spreadsheetTimeZone, 'MMMM'); // Format the deliverable date to get the month name
+	const year = Utilities.formatDate(deliverableDate, spreadsheetTimeZone, 'yyyy'); // Format the deliverable date to get the year
 	Logger.log(`Formatted month and year: ${month} ${year}`);
-
+  
+	// Calculate the exact deadline date based on submission window (e.g., in minutes or days)
+	const submissionWindowStart = new Date();
+	const submissionDeadline = new Date(submissionWindowStart.getTime() + SUBMISSION_WINDOW_MINUTES);  // Adjust as needed
+	const submissionDeadlineDate = Utilities.formatDate(submissionDeadline, spreadsheetTimeZone, 'MMMM dd, yyyy');
+  
 	emails.forEach((email, index) => {
-		const discordHandle = registrySheet.getRange(index + 2, 2).getValue(); // Fetch Discord Handle from column B
-		Logger.log(`Fetched Discord handle for email ${email}: ${discordHandle}`);
-
-		// Create the email message using the template and replace placeholders with actual values
-		const message = REQUEST_SUBMISSION_EMAIL_TEMPLATE.replace(
-			"{AmbassadorName}",
-			discordHandle
-		)
-			.replace("{Month}", month)
-			.replace("{Year}", year)
-			.replace("{SubmissionFormURL}", SUBMISSION_FORM_URL)
-			.replace("{SUBMISSION_WINDOW_MINUTES}", SUBMISSION_WINDOW_MINUTES);
-		Logger.log(`Email message created for ${email}: ${message}`);
-
-		if (!testing) {
-			MailApp.sendEmail(email, "☑️Request for Submission", message); // Send the email to the ambassador
-			Logger.log(`Email sent to ${email}`);
-		} else {
-			Logger.log(`Testing mode: Submission request email logged for ${email}`);
-		}
+	  const discordHandle = registrySheet.getRange(index + 2, 2).getValue();
+  
+	  // Prepare the email message with the deadline date
+	  const message = REQUEST_SUBMISSION_EMAIL_TEMPLATE
+		.replace('{AmbassadorName}', discordHandle)
```

Comment:
Should rename this token to AmbassadorDiscordHandle if it is for the handle, not the name.